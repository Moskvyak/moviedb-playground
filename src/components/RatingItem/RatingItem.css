.ratingitem {
    border-radius: 8px;
    font-size: 0.75em;
    padding: 4px 8px;
    font-weight: 600;
}