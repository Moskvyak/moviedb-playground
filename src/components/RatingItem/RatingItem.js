import React from 'react';

import './RatingItem.css';

const getRatingAndClass = (value) => {
    const rating = `${value * 10}%`;
    if (value < 4) {
        return { 
            rating,
            className: 'red'
        }
    }
    if (value < 7.5) {
        return {
            rating,
            className: 'purple'
        }
    }
    return {
        rating,
        className: 'primary'
    }
}

const RatingItem = React.memo(({ rating }) => {
    const ratingObject = getRatingAndClass(rating);
    const className = `${ratingObject.className}-background-color ratingitem`
    return <div className={className}>{ratingObject.rating}</div>
});

export default RatingItem;

