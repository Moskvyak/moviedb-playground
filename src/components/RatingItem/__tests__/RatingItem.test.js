import React from 'react';
import renderer from 'react-test-renderer';
import RatingItem from '../RatingItem';

describe('<RatingItem />', () => {
  it('should render with rating less than 4', () => {
    const component = renderer.create(<RatingItem rating={3.5} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render with rating between 4 and 7.5', () => {
    const component = renderer.create(<RatingItem rating={5} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render with rating more than 7.5', () => {
    const component = renderer.create(<RatingItem rating={8.5} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
