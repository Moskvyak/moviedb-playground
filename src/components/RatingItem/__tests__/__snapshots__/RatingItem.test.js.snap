// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingItem /> should render with rating between 4 and 7.5 1`] = `
<div
  className="purple-background-color ratingitem"
>
  50%
</div>
`;

exports[`<RatingItem /> should render with rating less than 4 1`] = `
<div
  className="red-background-color ratingitem"
>
  35%
</div>
`;

exports[`<RatingItem /> should render with rating more than 7.5 1`] = `
<div
  className="primary-background-color ratingitem"
>
  85%
</div>
`;
